ivy/functional/frontends/paddle/nn/functional/vision.py

```diff
@@ -135,3 +136,169 @@ def affine_grid(theta, out_shape, align_corners=True):
             base_grid[:, :, :, :, 3] = ivy.full((D, H, W), 1)
             grid = ivy.matmul(base_grid.view((N, D * H * W, 4)), theta.swapaxes(1, 2))
             return grid.view((N, D, H, W, 3))
+
+
+# grid_sample
+@to_ivy_arrays_and_back
+@with_unsupported_dtypes({"2.5.1 and below": ("float16", "bfloat16")}, "paddle")
+def grid_sample(input, grid, mode, padding_mode):
+    """
+    grid_sample is used to perfome various spatial transformations.
+
+    The function is used to resample the pixel values.
+    i.e from the original image to new positions defined by the grid of sampling points.
+
+    Basically, it's used to transform data from
+    one coordinate system to another.
+
+    Parameters:
+    - input: image to be transformed.
+    - grid: defines the transformation to be applied to the input tensor.
+    - mode: determines the interpolation method.
+    - padding_mode: controls how to handle out-of-bound values.
+
+    Returns:
+    Transformed tensor.
+    """
+
+    if mode not in ["nearest", "bilinear"]:
+        raise ValueError("Invalid mode. Supported modes are 'nearest' and 'bilinear'. ")
+
+    if padding_mode not in ["zeros", "border", "reflection"]:
+        raise ValueError(
+            "Invalid padding mode. Supported modes are 'zeros', 'border' and"
+            " 'reflection'. "
+        )
+
+    if len(grid.shape) != 4 or grid.shape[3] != 2:
+        raise ValueError(
+            "The grid should be a 4D tensor with the last dim having size 2."
+        )  # size 2 representing (x,y) coordinates.
+
+    if mode == "nearest":
+        return grid_sample_nearest(input, grid, padding_mode)
+    elif mode == "bilinear":
+        return grid_sample_bilinear(input, grid, padding_mode)
+
+
+def grid_sample_nearest(input, grid, padding_mode):
+    """
+    Assigns the pixel value at non-integer grid coordinates to be the value of the
+    nearest pixel of the input data. The transformation process involves normalising the
+    grid & rounding them to the nearest pixel value.
+
+    parameters:
+    - input: image that needs to be transformed.
+    - grid: the transformation grid.
+    - padding_mode: controls how to handle out-of-bound values.
+
+    Returns:
+    transformed values for each point in the output grid.
+    These transformed values represent the result of applying spatial transformation.
+    """
```

Comment:
please remove the docstring and rename the function to reflect it's private nature